src/stores/index.js

```diff
@@ -929,40 +834,120 @@ const Store = Reflux.createStore({
    *
    * @param {Object} connection - The current connection.
    */
-  _connect(connection) {
+  async _connect(connection) {
+    // Set the connection's app name to the electron app name of Compass.
+    connection.appname = electron.remote.app.getName();
+
     this.dataService = new DataService(connection);
     this.appRegistry.emit('data-service-initialized', this.dataService);
-    this.dataService.connect((error, ds) => {
-      if (error) {
-        this.StatusActions.done();
-        this.setState({
-          isValid: false,
-          errorMessage: error.message,
-          syntaxErrorMessage: null
-        });
-      } else {
-        const currentConnection = this.state.currentConnection;
-        const currentSaved = this.state.connections[currentConnection._id];
 
-        this.state.isValid = true;
-        this.state.isConnected = true;
-        this.state.errorMessage = null;
-        this.state.syntaxErrorMessage = null;
-        this.state.hasUnsavedChanges = false;
-        this.state.isURIEditable = false;
-        this.state.customUrl = this.state.currentConnection.driverUrl;
+    try {
+      const runConnect = promisify(this.dataService.connect.bind(this.dataService));
```

Comment:
Completely unrelated .. It would be so nice to have all of those methods promisified already in data-service and  connection-model not sure how could we make it happen though.

Reply:
Yes.